pkg/model/lattice/targetgroup.go

```diff
@@ -67,5 +135,50 @@ func NewTargetGroup(stack core.Stack, id string, spec TargetGroupSpec) *TargetGr
 
 	stack.AddResource(tg)
 
-	return tg
+	return tg, nil
+}
+
+func (t *TargetGroupTagFields) IsServiceExport() bool {
+	return t.K8SParentRefType == ParentRefTypeSvcExport
+}
+
+func (t *TargetGroupTagFields) IsRoute() bool {
+	return t.K8SParentRefType == ParentRefTypeHTTPRoute ||
+		t.K8SParentRefType == ParentRefTypeGRPCRoute
+}
+
+func (t *TargetGroupSpec) Validate() error {
+	requiredFields := []string{t.K8SServiceName, t.K8SServiceNamespace,
+		t.Protocol, t.ProtocolVersion, t.VpcId, t.EKSClusterName, t.IpAddressType,
+		string(t.K8SParentRefType)}
+
+	for _, s := range requiredFields {
+		if s == "" {
+			return errors.New("one or more required fields are missing")
+		}
+	}
+
+	if t.IsRoute() {
+		if t.K8SRouteName == "" || t.K8SRouteNamespace == "" {
+			return errors.New("route name or namespace missing for route-based target group")
+		}
+	}
+
+	return nil
+}
+
+func TgNamePrefix(spec TargetGroupSpec) string {
+	truncSvcNamespace := utils.Truncate(spec.K8SServiceNamespace, MaxNamespaceLength)
+	truncSvcName := utils.Truncate(spec.K8SServiceName, MaxNameLength)
+	return fmt.Sprintf("k8s-%s-%s", truncSvcNamespace, truncSvcName)
+}
+
+func GenerateTgName(spec TargetGroupSpec) string {
+	// tg max name length 128
+	prefix := TgNamePrefix(spec)
+	randomSuffix := make([]rune, RandomSuffixLength)
+	for i := range randomSuffix {
+		randomSuffix[i] = rune(rand.Intn(26) + 'a')
```

Comment:
so looks like the random suffix is made of alphabets, might have chance to hit some bad name here. One mitigation I heard before is to not use vowels